scripts/check_approximation_budget.py

```
@@ -0,0 +1,124 @@
+        return json.load(handle)
+
+
+def die(message: str) -> int:
+    print(f"error: {message}", file=sys.stderr)
+    return 1
+
+
+def require_number(mapping: Dict[str, Any], key: str, case_id: str, section: str) -> float:
+    value = mapping.get(key)
+    if not isinstance(value, (int, float)):
+        raise ValueError(f"{case_id}: {section}.{key} must be numeric")
```

Ruff (0.15.10)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)